src/abstractdataframe/abstractdataframe.jl

```diff
@@ -983,35 +992,78 @@ julia> vcat(df1, df2)
 │ 6   │ 6     │ 6     │
 ```
 """
-Base.vcat(df::AbstractDataFrame) = df
-Base.vcat(dfs::AbstractDataFrame...) = _vcat(collect(dfs))
-function _vcat(dfs::AbstractVector{<:AbstractDataFrame})
+Base.vcat(df::AbstractDataFrame; 
+          columns::Union{Symbol, AbstractVector{Symbol}}=:equal) = df
```

Comment:
This implementation is incorrect when `columns` is a vector.
Also in the future we will change this to always make a copy of columns (this is a separate PR).

Reply:
I added the subsetting if `columns` is a vector. Let's hope I get this in first and then you can rebase the ownership PR.